crates/fuel-core/src/graphql_api/database.rs

```diff
@@ -57,25 +57,77 @@ pub type OnChainView = Arc<dyn OnChainDatabase>;
 /// The off-chain view of the database used by the [`ReadView`] to fetch off-chain data.
 pub type OffChainView = Arc<dyn OffChainDatabase>;
 
+/// The GraphQL can't work with the generics in [`async_graphql::Context::data_unchecked`] and requires a known type.
+/// It is an `Arc` wrapper around the generic for on-chain and off-chain databases.
+struct ArcWrapper<Provider, ArcView> {
```

Comment:
maybe it would make sense to split this wrapper into a separate module so it doesn't clutter the main database implementation?